types/defines/ai.d.ts

```diff
@@ -100,16 +100,31 @@ export declare abstract class BaseAiTextEmbeddings {
   postProcessedOutputs: AiTextEmbeddingsOutput;
 }
 export type RoleScopedChatInput = {
-  role: "user" | "assistant" | "system" | "tool";
+  role: "user" | "assistant" | "system" | "tool" | string;
```

Comment:
`"user" | "assistant" | "system" | "tool" | string` is equivalent to just `string`.

I would recommend:
```suggestion
  role: "user" | "assistant" | "system" | "tool" | (string & {});
```

This is a well-used TypeScript quirk that allows any string value, but doesn't reduce the entire type to just `string`, therefore allowing autocomplete of the common roles.

Reply:
This is great, TIL, updated it in the new commit!